src/CodeGen_WebGPU_Dev.cpp

```diff
@@ -344,9 +404,20 @@ void CodeGen_WebGPU_Dev::CodeGen_WGSL::visit(const For *loop) {
         loop->body.accept(this);
 
     } else {
-        user_assert(loop->for_type != ForType::Parallel)
-            << "Cannot use parallel loops inside WebGPU shaders\n";
-        CodeGen_C::visit(loop);
+        user_assert(loop->for_type == ForType::Serial)
+            << "Cannot only use serial loops inside WebGPU shaders\n";
```

Comment:
The error message doesn't quite match the assert? Or maybe I misunderstand.

Reply:
Oops, fixed :-)